### PR TITLE
OpenCL-OpenGL interop support for apple

### PR DESCRIPTION
**Pull request by kallaballa**
## WIP

Experimental support for cl_APPLE_gl_sharing. Issue: #26273

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by kallaballa** [comment]
Would you please bump a Mac build?

File: modules/core/src/opengl.cpp [hunk 1]
```diff
@@ -1678,13 +1684,20 @@ Context& initializeContextFromGL()
...
         cl_context_properties properties[] =
         {
 #if defined(_WIN32)
             CL_CONTEXT_PLATFORM, (cl_context_properties)platforms[i],
             CL_GL_CONTEXT_KHR, (cl_context_properties)wglGetCurrentContext(),
             CL_WGL_HDC_KHR, (cl_context_properties)wglGetCurrentDC(),
-#elif defined(__ANDROID__)
+#elif defined (__APPLE__)
+            CL_CONTEXT_PROPERTY_USE_CGL_SHAREGROUP_APPLE, (cl_context_properties)cglShareGroup,
+#elif defined(__ANDROID__) || defined(MACOSX)
```
Comment (opencv-alalek) [inline modules/core/src/opengl.cpp:1700]: @asmorkalov Why Android?

Comment (kallaballa) [reply modules/core/src/opengl.cpp]: fixed

**Comment by kallaballa** [comment]
This PR is not satisfactory since it will never work on all required platforms. 
Closed in favor of: #26281

**Pull request closed without merging**